File: curso/src/main/java/br/ufmt/web/curso/repository/DisciplinaRepository.java
package br.ufmt.web.curso.repository;

import org.springframework.data.jpa.repository.support.JpaRepositoryImplementation;

import br.ufmt.web.curso.Disciplina;

public interface DisciplinaRepository
  extends JpaRepositoryImplementation
    <Disciplina,Long> {
  
}

File: cursodotnet/Models/DisciplinaRepository.cs
namespace Br.Ufmt.Web.Curso.Repository
{
  public interface IDisciplinaRepository : IRepository<Disciplina, long>
  { }

  public class DisciplinaRepository : Repository<Disciplina, long>, IDisciplinaRepository
  {
    public DisciplinaRepository(CursoContext context) : base(context)
    {
    }
  }
}